### DISCUSSION TITLE
v0.2.0

### DISCUSSION THREAD

**Discussion by jarmoza** [neurobagel/annotation_tool#480] [Announcements]
# What's Changed

### Overview

* Annotation tool now outputs a modified BIDS data dictionary that includes Neurobagel annotations section per participants.tsv column (In versions `0.1.x`, a fully transformed version of participants.tsv was outputted. See the description of `0.1.x` below for more details.) 
* Assessment tool categorization is temporarily removed until new version of assessment tool annotation is added
* Continuous value columns now can be annotated with the following transformation heuristics: bounded value, european decimal value, float value, and integer value
* Categorical value columns (currently Diagnosis and Sex) can now be annotated with a set of labels via Snomed and BIDS controlled vocabularies (see `categoricalOptions` in `store/index.js`)
* Vuex store has been refactored to accommodate all of the changes for this release
* Testing coverage has greatly increased. e2e and page-level tests have been refactored for the changes in this release, and there are now unit tests for all getters, mutations, and actions in the Vuex store
* Though the annotation tool's output is still accessible via the Download page, preview of the file's contents has been removed. (This will be implemented for the new data dictionary in an upcoming release.
* Updated figures, descriptions, install/deployment, and page instructions in README.md

### Contributors
* @jarmoza made their first contribution in https://github.com/neurobagel/annotation_tool/pull/5
* @surchs made their first contribution in https://github.com/neurobagel/annotation_tool/pull/27
* @rmanaem made their first contribution in https://github.com/neurobagel/annotation_tool/pull/241
* @alyssadai made their first contribution in https://github.com/neurobagel/annotation_tool/pull/279
* @yarikoptic made their first contribution in https://github.com/neurobagel/annotation_tool/pull/410
* @Remi-Gau made their first contribution in https://github.com/neurobagel/annotation_tool/pull/417
* @dependabot made their first contribution in https://github.com/neurobagel/annotation_tool/pull/418

### Changes

* Removal of nextPageAccessible getter by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/258
* Create LICENSE by @alyssadai in https://github.com/neurobagel/annotation_tool/pull/279
* Implemented mutation to change column to category mapping by @rmanaem in https://github.com/neurobagel/annotation_tool/pull/283
* Implement getCategories store getter by @surchs in https://github.com/neurobagel/annotation_tool/pull/284
* Implement getColumnDescription getter by @surchs in https://github.com/neurobagel/annotation_tool/pull/289
* Implement value description getter by @surchs in https://github.com/neurobagel/annotation_tool/pull/290
* createColumnToCategoryMap mutation + required getter + test by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/291
* Implemented getColumnNames getter  by @rmanaem in https://github.com/neurobagel/annotation_tool/pull/292
* setCurrentPage mutation + unit test for store refactor by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/296
* Implementation of nextPage getter for refactor + unit test by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/297
* index page refactor by @surchs in https://github.com/neurobagel/annotation_tool/pull/302
* Implementation of isPageAccessible getter for refactor by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/299
* action to process a new dataTable by @surchs in https://github.com/neurobagel/annotation_tool/pull/303
* Data dictionary refactor for Store refactor + Tests by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/312
* Update to Vuex syntax for getters and unit tests by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/321
* Components and their tests are nominally refactored for store by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/323
* Mutations should have a single payload parameter by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/325
* Swapping in refactored store by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/326
* Basic app launch post-refactor by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/327
* Navbar/next page button/page accessibility update for refactor by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/328
* Updates to Index page for data table + data dictionary for store refactor by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/329
* Adjusting e2e tests + support code for Phase 1 store refactor by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/330
* Build static by @surchs in https://github.com/neurobagel/annotation_tool/pull/335
* New mutation setHeuristic for phase 2 store refactor by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/337
* New 'getHeuristic' getter for phase 2 store refactor by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/342
* Only run index & categ e2e tests for now by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/343
* Refactor getter 'getAnnotationComponent' by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/345
* getMappedColumns Getter for Phase 2 Store Refactor by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/349
* New getter for category explanation text for the phase 2 store refactor by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/351
* getOptions getter for Phase 2 store refactor by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/353
* Add node v16 dependency info by @surchs in https://github.com/neurobagel/annotation_tool/pull/356
* Getter getMissingValues for Phase 2 Store Refactor by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/355
* Missing values table items refactor + component test fix by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/357
* mutation 'changeMissingStatus' for Phase 2 Store Refactor by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/359
* Added and configured lint staged and lint workflow by @rmanaem in https://github.com/neurobagel/annotation_tool/pull/362
* Added and configured cd workflow by @rmanaem in https://github.com/neurobagel/annotation_tool/pull/361
* getUniqueValues getter for Phase 2 Store Refactor by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/363
* changeMissingStatus getter Fix + Check for Unique Values by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/365
* selectCategoricalOption mutation for Phase 2 Store Refactor by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/366
* e2e node version update by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/370
* getHarmonizedPreview getter for Phase 2 Refactor by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/368
* Feature for Phase 2 Refactor: Transform Heuristics for continuous values are now selectable by column by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/372
* Refactor: handle controlled terms as objects by @surchs in https://github.com/neurobagel/annotation_tool/pull/374
* Phase 2 Store Refactor - Integration Part 1 - Annotation page access, tabs refactor, column-category unlinking by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/375
* Phase 2 refactor: New getter getSelectedCategoricalOption by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/378
* Phase 2 refactor: Column linking/unlinking alters data dictionary by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/379
* Replace remaining getSelectedOption instances and fix optional chaining bug with new getter by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/380
* Phase 2 Refactor: Missing values functionality + integration with Annotation page by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/382
* Old tool group component removal, old store removal, and store field rename by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/383
* Implemented `updateAnnotationCount` mutation by @rmanaem in https://github.com/neurobagel/annotation_tool/pull/389
* Capitalized references of `description` in store and updated affected unit tests by @rmanaem in https://github.com/neurobagel/annotation_tool/pull/391
* Fixed the bug that made download page unreachable from annotation page by @rmanaem in https://github.com/neurobagel/annotation_tool/pull/396
* Bumped cypress version to `12.9.0` by @rmanaem in https://github.com/neurobagel/annotation_tool/pull/397
* Categorical Select bugfix + Rework of initialization of data dictionary columns for data types by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/395
* Removed `range` and `string` options from `transformationHeuristics` by @rmanaem in https://github.com/neurobagel/annotation_tool/pull/401
* Refactored download button by @rmanaem in https://github.com/neurobagel/annotation_tool/pull/404
* Removal of old download components and related functionality by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/409
* Refactored e2e tests by @rmanaem in https://github.com/neurobagel/annotation_tool/pull/405
* codespell: workflow, config, typos fixed by @yarikoptic in https://github.com/neurobagel/annotation_tool/pull/410
* [INFRA] add dependabot by @Remi-Gau in https://github.com/neurobagel/annotation_tool/pull/417
* Next page button component tests - Clean branch by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/414
* [ENH] Updated `Diagnosis` options by @rmanaem in https://github.com/neurobagel/annotation_tool/pull/428
* Getter for json output for discrete (categorical) value columns by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/426
* [ENH] Implemented getter to create JSON representation of continuous column by @rmanaem in https://github.com/neurobagel/annotation_tool/pull/430
* [ENH] Meta-getter for JSON output from download page by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/431
* [ENH] Restore download functionality for refactor/new meta getter by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/433
* [ENH] Merging dev branch into master post-refactor by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/437
* Updated cd.yaml by @rmanaem in https://github.com/neurobagel/annotation_tool/pull/440
* Bump codespell-project/actions-codespell from 1 to 2 by @dependabot in https://github.com/neurobagel/annotation_tool/pull/418
* Bump actions/stale from 6 to 8 by @dependabot in https://github.com/neurobagel/annotation_tool/pull/419
* Bump terser from 4.8.0 to 4.8.1 by @dependabot in https://github.com/neurobagel/annotation_tool/pull/421
* Bump ua-parser-js from 0.7.31 to 0.7.35 by @dependabot in https://github.com/neurobagel/annotation_tool/pull/422
* Bump loader-utils from 1.4.0 to 1.4.2 by @dependabot in https://github.com/neurobagel/annotation_tool/pull/448
* Bump parse-path and @nuxt/telemetry by @dependabot in https://github.com/neurobagel/annotation_tool/pull/423
* Bump nth-check and nuxt by @dependabot in https://github.com/neurobagel/annotation_tool/pull/449
* Bump json5 from 1.0.1 to 1.0.2 by @dependabot in https://github.com/neurobagel/annotation_tool/pull/450
* Bump decode-uri-component from 0.2.0 to 0.2.2 by @dependabot in https://github.com/neurobagel/annotation_tool/pull/451
* Bump follow-redirects from 1.14.7 to 1.15.2 by @dependabot in https://github.com/neurobagel/annotation_tool/pull/452
* [MAINT] Instructed dependabot to ignore nuxt and vue major updates by @rmanaem in https://github.com/neurobagel/annotation_tool/pull/455
* Bump papaparse from 5.3.1 to 5.4.1 by @dependabot in https://github.com/neurobagel/annotation_tool/pull/458
* Bump cy-verify-downloads from 0.1.8 to 0.1.14 by @dependabot in https://github.com/neurobagel/annotation_tool/pull/456
* Bump eslint-plugin-prettier from 4.0.0 to 4.2.1 by @dependabot in https://github.com/neurobagel/annotation_tool/pull/459
* [ENH] Updates to README for descriptions, how-to-use instructions, screenshots, and badges by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/447
* Bump bootstrap-vue from 2.21.2 to 2.23.1 by @dependabot in https://github.com/neurobagel/annotation_tool/pull/463
* Bump eslint from 8.11.0 to 8.43.0 by @dependabot in https://github.com/neurobagel/annotation_tool/pull/462
* Bump cypress from 12.9.0 to 12.14.0 by @dependabot in https://github.com/neurobagel/annotation_tool/pull/461
* Bump husky from 7.0.4 to 8.0.3 by @dependabot in https://github.com/neurobagel/annotation_tool/pull/460
* [MAINT] Instructed dependabot to ignore webpack major version updates by @rmanaem in https://github.com/neurobagel/annotation_tool/pull/465
* Bump editorconfig from 0.15.3 to 1.0.2 by @dependabot in https://github.com/neurobagel/annotation_tool/pull/470
* Bump bootstrap from 4.6.1 to 5.3.0 by @dependabot in https://github.com/neurobagel/annotation_tool/pull/469
* Bump lint-staged from 13.1.2 to 13.2.2 by @dependabot in https://github.com/neurobagel/annotation_tool/pull/467
* Bump vue-select from 3.18.3 to 3.20.2 by @dependabot in https://github.com/neurobagel/annotation_tool/pull/466

**Full Changelog**: https://github.com/neurobagel/annotation_tool/commits/0.2.0

### Contributors
* @jarmoza made their first contribution in https://github.com/neurobagel/annotation_tool/pull/5
* @surchs made their first contribution in https://github.com/neurobagel/annotation_tool/pull/27
* @rmanaem made their first contribution in https://github.com/neurobagel/annotation_tool/pull/241

### Changes

* Ability to upload file and store it as data via landing page by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/5
* Example data by @surchs in https://github.com/neurobagel/annotation_tool/pull/27
* Landing page file loading, Clickable columns on Categorization Page, and Vuex Store by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/28
* Blank annotation for mockup image by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/33
* Navigation and state by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/42
* move examples to their own root dir by @surchs in https://github.com/neurobagel/annotation_tool/pull/54
* Transparent category colors and some related UI polish for the categorization page by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/47
* Page state preservation for Categorization page, Palette/Table clicking simplification, and general cleanup for landing and categorization pages by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/61
* New global store version + new dynamic styling for categorization page by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/66
* [STYLE] add ESLint and prettier by @surchs in https://github.com/neurobagel/annotation_tool/pull/75
* [FEAT] add explanation component by @surchs in https://github.com/neurobagel/annotation_tool/pull/76
* [FEAT] display categorized columns on the annotation page by @surchs in https://github.com/neurobagel/annotation_tool/pull/77
* Initial work to integrate annotation page components and global store by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/84
* Creating a layout for the tool and moving navbar to the layout by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/86
* [FEAT] add discrete value annotation by @surchs in https://github.com/neurobagel/annotation_tool/pull/87
* Prevents blank lines from tsv data passing through the annotation process by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/90
* Adding download page and hooking it into annotation tool workflow by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/91
* [FEAT] add vocabulary annotation by @surchs in https://github.com/neurobagel/annotation_tool/pull/92
* [FIX] pass the original and annotated dataTable to components by @surchs in https://github.com/neurobagel/annotation_tool/pull/97
* Changing annotation table status check to see if it differs from original table by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/98
* add example dataset to illustrate transformations by @surchs in https://github.com/neurobagel/annotation_tool/pull/99
* Styling improvements for demo by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/100
* Post demo cleanup - Annotation page component overhaul and formatting by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/108
* Initial missing values implementation by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/110
* Implement missing values component by @surchs in https://github.com/neurobagel/annotation_tool/pull/123
* Implement missing values discrete by @surchs in https://github.com/neurobagel/annotation_tool/pull/132
* Fixes for unique explanation component ID and client-only tags by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/145
* Assessment Tool Groups UI on Categorization Page by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/144
* Assessment tool group tabs and enhanced functionality by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/150
* Implement missing values vocabulary by @surchs in https://github.com/neurobagel/annotation_tool/pull/143
* Precommit hooks via husky by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/156
* Missing value functionality for assessment tool groups + related changes by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/159
* Annotation tool now outputs JSON files reflecting the annotated table by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/163
* First Cypress End to End Testing Implementation by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/165
* Remove the old mockup image by @surchs in https://github.com/neurobagel/annotation_tool/pull/168
* Initial page-specific tests for the Annotation page by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/182
* Update output format to data model by @surchs in https://github.com/neurobagel/annotation_tool/pull/191
* Fix for categorical double save bug on annotation page by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/196
* File select cancellation fix + condition cleanup by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/197
* Explanation component test added by @surchs in https://github.com/neurobagel/annotation_tool/pull/219
* Refactor and test missing value component by @surchs in https://github.com/neurobagel/annotation_tool/pull/222
* Split cypress test actions by @surchs in https://github.com/neurobagel/annotation_tool/pull/226
* Continuous value component refactor by @surchs in https://github.com/neurobagel/annotation_tool/pull/224
* Refactor of category-select-table component by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/229
* Column linking table refactor + component test by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/230
* Refactored annot-categorical component so it's ready for the store by @rmanaem in https://github.com/neurobagel/annotation_tool/pull/241
* Refactored annot-columns component so it's ready for the store by @rmanaem in https://github.com/neurobagel/annotation_tool/pull/249
* Next page accessibility/navbar/layout refactor + All page refactor by @jarmoza in https://github.com/neurobagel/annotation_tool/pull/233
* Refactor annotation-tab component  by @surchs in https://github.com/neurobagel/annotation_tool/pull/242

<hr /><em>This discussion was created from the release <a href='https://github.com/neurobagel/annotation_tool/releases/tag/0.2.0'>v0.2.0</a>.</em>

**Comment by Remi-Gau** [comment]
Was this announcement posted automatically?
